### PR TITLE
Add getcurrentappenvironment

### PR DESCRIPTION
**Pull request by dimaportenko**
## Description

Related #1216

add `getCurrentAppEnvironment` function for ios to determine whether app was downloaded from `AppStore`,`TestFlight` or `Other`

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ❌     |
| Windows |    ❌     |

## Checklist

* [x] I have tested this on a device/simulator for each compatible OS
* [ ] I added the documentation in `README.md`
* [ ] I mentioned this change in `CHANGELOG.md`
* [x] I updated the typings files (`privateTypes.ts`, `types.ts`)
* [ ] I added a sample use of the API (`example/App.js`)

### CODE REVIEW

File: src/index.js.flow [hunk 1]
```diff
@@ -44,6 +46,8 @@ declare module.exports: {
   getCarrierSync: () => string,
   getCodename: () => Promise<string>,
   getCodenameSync: () => string,
+  getCurrentAppEnvironment: () => Promise<string>,
+  getCurrentAppEnvironmentSync: () => string,
```
Comment (schie) [inline src/index.js.flow:50]: Wouldn't these return the `AppEnvironment` type?

File: src/internal/privateTypes.ts [hunk 2]
```diff
@@ -55,6 +55,8 @@ interface ExposedNativeMethods {
   getCarrierSync: () => string;
   getCodename: () => Promise<string>;
   getCodenameSync: () => string;
+  getCurrentAppEnvironment: () => Promise<string>;
+  getCurrentAppEnvironmentSync: () => string;
```
Comment (schie) [inline src/internal/privateTypes.ts:59]: Shouldn't these return the `AppEnvironment` type?

**Comment by mikehardy** [comment]
I would like for this interface to be unified with the existing similar interface for android that (and windows) that returns package installer. Yes the types will be different but I assume they are different for windows and android as well (they should be...) yet at heart they are all strings, and if the strings are formally specified (i.e. constrained to a small set as the values you mention above) that can still be unioned in type-wise

I don't think this requires a completely new API

so to be specific - please just make this the iOS implementation of the existing API which is not currently supported on iOS and I think we're close. I love that seeing the app install method for iOS will be possible with this though! that can b very useful for feature toggles etc (like, showing a bug report / feedback link or not, turning on screen recording during usage or not etc)

**Comment by dimaportenko** [comment]
@mikehardy so you would like instead of the new API function it'll be iOS implementation of `getInstallerPackageName()`, is it correct?

**Comment by mikehardy** [comment]
Exactly that, and thank you! As I said I can imagine this being really useful for people

**Comment by dimaportenko** [comment]
@mikehardy ok, it makes sense to me as well. I'll do this change.

**Comment by dimaportenko** [comment]
I changed the implementation. It's `getInstallerPackageName()` for iOS now. Please take a look.

**Comment by schie** [comment]
@dimaportenko can you update the README to reflect your changes? 

Otherwise, it looks good to me.

**Comment by dimaportenko** [comment]
@schie done

**Review by mikehardy — APPROVED** [review]
Nice this looks excellent to me as well, thank you @dimaportenko !

**Comment by schie** [comment]
hmm.. looking now

**Comment by schie** [comment]
I need to look more into RNDI's permissions

**Pull request merged**